store/httpstore.go

```diff
@@ -99,10 +99,6 @@ func (s HTTPStore) GetMeta(name string, size int64) ([]byte, error) {
 	logrus.Debugf("%d when retrieving metadata for %s", resp.StatusCode, name)
 	b := io.LimitReader(resp.Body, size)
 	body, err := ioutil.ReadAll(b)
-	if resp.ContentLength > 0 && int64(len(body)) < resp.ContentLength {
```

Comment:
@mtrmac @diogomonica @NathanMcCauley @cyli I realized that this will cause a failure whenever we download a timestamp (which uses a 5MB max size). We've been getting away with it because the notary server doesn't currently return the Content-Length header. 

Given how TUF uses the size for timestamp, I think it's more correct to treat size as an upper limit so I'm removing this check and error. What are your thoughts?